src/solidity-frontend/solidity_convert.cpp

```diff
@@ -4073,6 +4304,103 @@ nlohmann::json solidity_convertert::make_array_elementary_type(
   return elementary_type;
 }
 
+std::vector<std::string> splitByDelimiter(const std::string &str, const std::string &delimiter)
+{
+  std::vector<std::string> output;
+  size_t startPos = 0;
+  size_t endPos;
+  std::string token;
+  
+  while ((endPos = str.find(delimiter, startPos)) != std::string::npos) {
+      // endPos updates itself every iteration if delimiter is present
+      token = str.substr(startPos, endPos - startPos);
+      if (!token.empty()) {
+          output.push_back(token);
+      }
+      // startPos will start from the part of the string where
+      // the delimiter is removed
+      startPos = endPos + delimiter.length();
+  }
+  // Add the last part
+  token = str.substr(startPos);
+  if (!token.empty()) {
+      output.push_back(token);
+  }
+
+  return output;
+}
+
+std::vector<nlohmann::json> solidity_convertert::make_struct_elementary_types(
```

Comment:
I think you can obtain the type info in the `typeDescriptions` node in each component, right? 

You can do something like:
```c++
   if (get_expr(args.at(i), members.at(i)["typeDescriptions"], init))
        return true;
```

Reply:
I have yet to delete this function in case I may use it in the future for an unknown reason. I will delete it shortly